packages/frontEnd/src/slice/login.tsx

```diff
@@ -13,27 +14,42 @@ interface ResponseProps {
 interface LoginState {
   status: "waiting" | "success" | "pending" | "error";
   token: null | string;
+  errorMessage: null | string;
 }
 
-export const signIn = createAsyncThunk(
-  "auth/login",
-  async ({ email, password }: LoginProps, { rejectWithValue }) => {
-    try {
-      const response = await AuthService.login({
-        email,
-        password,
-      });
-      return response.data as ResponseProps;
-    } catch (error) {
-      const errorObject = error as Error;
-      return rejectWithValue(errorObject.message);
+interface ValidationErrors {
+  error: string;
+}
+
+export const signIn = createAsyncThunk<
+  ResponseProps,
+  LoginProps,
+  { rejectValue: ValidationErrors }
+>("auth/login", async ({ email, password }, { rejectWithValue }) => {
+  try {
+    const response = await AuthService.login({
+      email,
+      password,
+    });
+    localStorage.setItem("token", response.data.token);
+    return response.data;
+  } catch (err) {
+    const error = err as AxiosError<ValidationErrors>;
+    if (!error.response) {
+      throw err;
     }
+    return rejectWithValue(error.response.data);
   }
```

Comment:
don't cast error to an AxiosError
use a typeguard from axios `isAxiosError`, typescript then knows what's the underlying type :))
```suggestion
  } catch (error) {
    if (isAxiosError(error)) {
      return rejectWithValue(error.response.data);
    }
    rejectWithValue("Unhandled error")
  }
```

Reply:
What is the benefit of your suggestion? I still have to use AxiosError to make it work.

Reply:
I see, this suggestion is not good here because much more work needs to be done to have this working better

But answering your question asserting variables is bad because you are not always sure if this value is an AxiosError<ValidationErrors>. It would be better to use a typeguard. But if you don't want to leave it how it's currently :D

Reply:
Ok, thanks for answering. I'm going to leave it as it is.